frontend/src/features/userProfile/UserProfileDatasetList.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+
+import ContentBox from '../../chrome/ContentBox'
+import PageControl from '../../chrome/PageControl'
+import DropdownMenu from '../../chrome/DropdownMenu'
+import DatasetList from '../../chrome/DatasetList'
+import Icon from '../../chrome/Icon'
+import {
+  UserProfileParams,
+  CleanUserProfileParams
+} from '../../qri/userProfile'
+import { PaginatedResults } from './state/userProfileState'
+import { PageChangeObject } from '../../chrome/PageControl'
+
+export interface UserProfileDatasetListProps {
+  paginatedResults: PaginatedResults
+  userProfileParams: UserProfileParams
+  onParamsUpdate: (newQueryParams: UserProfileParams) => void
+}
+
+const UserProfileDatasetList: React.FC<UserProfileDatasetListProps> = ({
+  paginatedResults,
+  userProfileParams,
+  onParamsUpdate
+}) => {
+  const {
+    results,
+    pageInfo,
+  } = paginatedResults
+
+  const {
+    page,
+  } = pageInfo
+
+  const {
+    sort
+  } = userProfileParams
+
+  const menuItems = [
+    {
+      text: 'Dataset Name',
+      active: sort === 'name',
+      onClick: () => { onParamsUpdate({ sort: 'name' }) }
+    },
+    {
+      text: 'Recently Updated',
+      active: sort === 'recentlyupdated',
+      onClick: () => { onParamsUpdate({ sort: 'recentlyupdated' }) }
+    }
+  ]
+
+  const totalPages = Math.ceil( pageInfo.resultCount / pageInfo.pageSize )
+
+  // handle page change from PageControl
+  const handlePageChange = ({ selected: pageIndex }: PageChangeObject) => {
+    onParamsUpdate({ page: pageIndex + 1 })
+  }
+
+  return (
+    <>
+      <ContentBox className='mb-6 rounded-tl-none pt-5 pb-5 pr-5 pl-5'>
+        <div className='flex items-center justify-between border-b pb-5'>
+          <div className='text-qrigray'>
+            Page {page} of {totalPages}
+          </div>
+          <DropdownMenu items={menuItems} className='ml-8'>
+            <div className='border border-qrigray-300 rounded-lg text-qrigray-400 text-xs font-normal px-2 py-2 cursor-pointer'>
+              Sort By
+              <Icon icon='caretDown' size='2xs' className='ml-3' />
+            </div>
+          </DropdownMenu>
+        </div>
+        <DatasetList datasets={results} />
+      </ContentBox>
+      <PageControl
+        pageInfo={pageInfo}
+        queryParams={CleanUserProfileParams(userProfileParams)}
```

Comment:
the only thing I'm confused about is why we are pulling out default values from the userProfileParams

Reply:
wait i think i get it now, after seeing how this is handled in the `Search` page. This is sanitizing what the users see in the url, not what we use to actually fetch from the backend, correct? So if the requests are the "default" we aren't crowding the url.

Reply:
yep, this just keeps the URL free of default values.